File: qe-selenium/qe-selenium-java/src/main/java/com/demos/Activity13.java
package com.demos;

import java.util.List;

import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;

public class Activity13 {
    public static void main(String[] args) {
        FirefoxDriver driver=new FirefoxDriver();
        driver.get("https://training-support.net/webelements/tables/");
        driver.getTitle();
        List<WebElement> rows =driver.findElements(By.xpath("//table/tbody/tr"));
        List<WebElement> columns=driver.findElements(By.xpath("//table/tbody/tr[1]/td"));
        // List<WebElement> s_row=driver.findElements(By.xpath("//table/tbody/tr[2]/td[2]"));
        //  String cellValue =
        //         driver.findElement(By.xpath("//table/tbody/tr[2]/td[2]"))
        //               .getText();

        // System.out.println("Cell value (2,2): " + cellValue);

        System.out.println("Number of rows: " + rows.size());
        System.out.println("Number of column: " + columns.size());
        System.out.println("Cell value (2,2):" + driver.findElement(By.xpath("//table/tbody/tr[2]/td[2]")).getText());
        driver.quit();

    }
}

File: qe-selenium/qe-selenium-python/Activity13.py
from selenium import webdriver
from selenium.webdriver import Firefox
from selenium.webdriver.common.action_chains import ActionChains
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC


with webdriver.Firefox() as driver:
    driver.get("https://training-support.net/webelements/tables/")
    print(driver.title)
    rows=driver.find_elements(By.XPATH,"/html/body/div/main/div/div/div/div/div[2]/table/tbody/tr")
    columns=driver.find_elements(By.XPATH,"/html/body/div/main/div/div/div/div/div[2]/table/tbody/tr[1]/td")
    print(len(rows),len(columns))
    driver.quit
